src/server/scripts/Northrend/CrusadersColiseum/TrialOfTheChampion/boss_argent_challenge.cpp

```diff
@@ -152,40 +175,35 @@ class boss_eadric : public CreatureScript
 {
 public:
     boss_eadric() : CreatureScript("boss_eadric") { }
-    struct boss_eadricAI : public ScriptedAI
+
+    struct boss_eadricAI : public BossAI
     {
-        boss_eadricAI(Creature* creature) : ScriptedAI(creature)
+        boss_eadricAI(Creature* creature) : BossAI(creature, BOSS_ARGENT_CHALLENGE_E)
         {
             Initialize();
-            instance = creature->GetInstanceScript();
             creature->SetReactState(REACT_PASSIVE);
             creature->SetFlag(UNIT_FIELD_FLAGS, UNIT_FLAG_NON_ATTACKABLE);
         }
 
         void Initialize()
         {
-            uiVenganceTimer = 10000;
-            uiRadianceTimer = 16000;
-            uiHammerJusticeTimer = 25000;
-            uiResetTimer = 5000;
-
-            bDone = false;
+            _defeated = false;
         }
 
-        InstanceScript* instance;
-
-        uint32 uiVenganceTimer;
-        uint32 uiRadianceTimer;
-        uint32 uiHammerJusticeTimer;
-        uint32 uiResetTimer;
-
-        bool bDone;
-
         void Reset() override
         {
+            _Reset();
             Initialize();
         }
 
+        void SpellHit(Unit* /*caster*/, const SpellInfo *spell) override
+        {
+            if (IsHeroic())
+                if (spell->Id == SPELL_HAMMER_THROW_BACK)
+                    if (me->GetHealth() <= (uint32)spell->GetEffect(0)->BasePoints)
```

Comment:
What do you mean? I must remove "if (IsHeroic())" and magic number in ... spell->GetEffect(EFFECT_0)